pkg/clients/metal.go

```diff
@@ -113,3 +119,15 @@ func IsNotFound(err error) bool {
 	}
 	return false
 }
+
+// IsAlreadyDone returns true if, during VLAN assignment operations, the API
+// returns an error like "422 Virtual network 1182 already assigned" or "422
+// Virtual network 1182 already unassigned"
+func IsAlreadyDone(err error) bool {
+	if e, ok := err.(*packngo.ErrorResponse); ok && e.Response != nil {
+		return e.Response.StatusCode == http.StatusUnprocessableEntity &&
```

Comment:
Is status code 422 enough to identify this error?

Reply:
Good catch.

I am also checking the text included in the error message (the bold elements are being checked):

---

**422** Unprocessable Entity
{"error":"**Virtual Network** 1234 **already** attached"}

---

This will also capture "**Virtual Network** 1234 **already** unattached"
(I was originally checking for the text in the wrong place, StatusCode vs errsInOne. Fixed in the latest commits).